test/mlir/onnx/onnx_location.mlir

```diff
@@ -0,0 +1,26 @@
+
+// RUN: onnx-mlir --EmitMLIR  --preserveLocations --printIR %s |  FileCheck %s ; rm %p/*.onnx.mlir ; rm %p/*.tmp
+
+module {
+  func @main_graph(%arg0: tensor<1x16xf32>, %arg1: tensor<1x16xf32>, %arg2: tensor<1x16xf32>) -> tensor<1x16xf32> attributes {input_names = ["X", "Y", "U"], output_names = ["Z"]} {
+    %0 = "onnx.Add"(%arg0, %arg1) : (tensor<1x16xf32>, tensor<1x16xf32>) -> tensor<1x16xf32> loc("/build/workspace/addop.onnx":1:0)
+    %1 = "onnx.Add"(%0, %arg2) : (tensor<1x16xf32>, tensor<1x16xf32>) -> tensor<1x16xf32> loc("/build/workspace/addop.onnx":2:0)
+    return %1 : tensor<1x16xf32>
+  }
+  "onnx.EntryPoint"() {func = @main_graph, numInputs = 3 : i32, numOutputs = 1 : i32} : () -> ()
+
+// CHECK: loc("{{(/[[:alnum:]]+)+}}.onnx":1:0)
+}
+
+// RUN: onnx-mlir --EmitMLIR  %s | FileCheck %s ; rm %p/*.onnx.mlir ; rm %p/*.tmp
```

Comment:
Move this RUN line to the top and add a check prefix:

```
.... | FileCheck %s --check-prefix=SOME_NAME_HERE
```

then your line for checking things looks like this:

```
// SOME_NAME_HERE-NOT: loc("{{(/[[:alnum:]]+)+}}.onnx":1:0)
```

You can then delete the second `func` since it's a duplicate of the first.

You should also remove the `module`, there is a going to be a default module for this file.